test/functional/eval/input_spec.lua

```diff
@@ -196,6 +228,19 @@ describe('input()', function()
     eq('Vim(call):E118: Too many arguments for function: input',
        exc_exec('call input("prompt> ", "default", "file", "extra")'))
   end)
+  it('supports highlighting', function()
+    feed([[:call input({"highlight": "RainBowParens"})<CR>]])
+    wait()
+    feed('(())')
+    wait()
```

Comment:
I was referring to the extra `wait()`,  which will not be needed if remove the narrow width issue (make the window wider, or use `command`)

I've also noticed that the cmdline redraw logic is not very efficient. We needed some throttling to implement the cursor type switching without overloading the ui protocol serialization with data....